### PR TITLE
[search] Merged Processor and ProcessorV2.

### PR DESCRIPTION
**Pull request by mpimenov**

### CODE REVIEW

**Comment by mpimenov** [comment]
DUDE JTTP

**Comment by mapsmetest** [comment]
Build started sha1 is merged.

**Comment by mapsmetest** [comment]
GitHub Watchdog Success

**Comment by mpimenov** [comment]
PTAL

File: search/processor.cpp [hunk 1]
```diff
@@ -431,98 +494,40 @@ void Processor::ProcessEmojiIfNeeded(strings::UniString const & token, size_t in
...
-      subTokens.clear();
-      SplitUniString(token, MakeBackInsertFunctor(subTokens), delims);
-      if (subTokens.empty())
-        continue;
-
-      if (numHashes == 1)
-        m_tokens.push_back(strings::MakeUniString("#") + subTokens[0]);
-      else
-        m_tokens.emplace_back(move(subTokens[0]));
+  return search::GetQueryIndexScale(viewport);
```
Comment (ygorshenin) [inline search/processor.cpp]: Do we really need this short function?

File: search/processor.cpp [hunk 2]
```diff
@@ -431,98 +494,40 @@ void Processor::ProcessEmojiIfNeeded(strings::UniString const & token, size_t in
...
+    SuggestStrings(results);
 
-  // Assign prefix with last parsed token.
-  if (!m_tokens.empty() && !delims(strings::LastUniChar(query)))
-  {
-    m_prefix.swap(m_tokens.back());
-    m_tokens.pop_back();
-  }
+  v2::Geocoder::Params params;
+  InitParams(false /* localitySearch */, params);
```
Comment (ygorshenin) [inline search/processor.cpp]: InitParams is always called with false as it's first argument, do we really need it?

File: search/processor.cpp [hunk 3]
```diff
@@ -431,98 +494,40 @@ void Processor::ProcessEmojiIfNeeded(strings::UniString const & token, size_t in
...
+  m_geocoder.GoEverywhere(m_preRanker);
 
-  // get preffered types to show in results
-  m_prefferedTypes.clear();
-  ForEachCategoryTypes(v2::QuerySliceOnRawStrings<decltype(m_tokens)>(m_tokens, m_prefix),
-                       [&](size_t, uint32_t t)
-                       {
-                         m_prefferedTypes.insert(t);
-                       });
+  FlushResults(params, results, false /* allMWMs */, limit, false /* oldHouseSearch */);
```
Comment (ygorshenin) [inline search/processor.cpp]: oldHouseSearch is unused.

File: search/processor_factory.hpp [hunk 4]
```diff
@@ -1,7 +1,7 @@
 #pragma once
 
 #include "search/suggest.hpp"
-#include "search/v2/processor_v2.hpp"
+#include "search/processor.hpp"
```
Comment (ygorshenin) [inline search/processor_factory.hpp]: Sort, please.

File: search/processor.cpp (hunk 3 above)
Comment (ygorshenin) [inline search/processor.cpp]: FlushResults is always called with false for allMWMs, do we need this param too?

File: search/processor.cpp [hunk 5]
```diff
@@ -244,6 +247,138 @@ Processor::Processor(Index & index, CategoriesHolder const & categories,
...
+  SetInputLocale(locale);
+
+#ifdef FIND_LOCALITY_TEST
+  m_locality.SetLanguage(code);
+#endif
+}
+
+void Processor::SetInputLocale(string const & locale)
+{
+  if (!locale.empty())
```
Comment (ygorshenin) [inline search/processor.cpp]:
```c++
if (locale.empty())
  return;
...
```

**Comment by mpimenov** [comment]
DUDE JTTP

File: search/processor.cpp (hunk 1 above)
Comment (mpimenov) [inline search/processor.cpp]: Removed from processor but not from search/geometry_utils.

File: search/processor.cpp (hunk 2 above)
Comment (mpimenov) [inline search/processor.cpp]: I don't think so. The locality-search code belongs to geocoder anyway.
done

File: search/processor.cpp (hunk 3 above)
Comment (mpimenov) [inline search/processor.cpp]: Removed both.

File: search/processor_factory.hpp (hunk 4 above)
Comment (mpimenov) [inline search/processor_factory.hpp]: done

File: search/processor.cpp (hunk 5 above)
Comment (mpimenov) [inline search/processor.cpp]: done

**Comment by mapsmetest** [comment]
Build started sha1 is merged.

**Comment by mapsmetest** [comment]
GitHub Watchdog Error

**Comment by mpimenov** [comment]
PTAL

**Comment by therearesomewhocallmetim** [comment]
DUDE JTTP

**Comment by mapsmetest** [comment]
Build started sha1 is merged.

**Comment by mapsmetest** [comment]
GitHub Watchdog Failure

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**